include/lemon/tmalign.hpp

```diff
@@ -288,50 +25,74 @@ inline size_t find_element_by_name(const chemfiles::Frame& frame,
     return frame.size();
 }
 
-inline size_t find_operlapping_residues(const chemfiles::Frame& search,
-                                        const chemfiles::Frame& native,
-                                        std::vector<size_t>& a_search,
-                                        std::vector<size_t>& a_native) {
+inline unsigned long find_operlapping_residues(const chemfiles::Frame& search,
+                                               const chemfiles::Frame& native,
+                                               std::vector<size_t>& a_search,
+                                               std::vector<size_t>& a_native,
+                                               const std::string& s_chain,
+                                               const std::string& n_chain) {
+
+    // Setup which residues to return
     const auto& search_res = search.topology().residues();
     const auto& native_res = native.topology().residues();
 
     a_search.reserve(search_res.size());
     a_native.reserve(native_res.size());
 
-    size_t n_ali = 0;
-    for (const auto& sres : search_res) {
-        auto i = find_element_by_name(search, sres, "CA");
+    auto n_ali = 0ul;
+    for (const auto& s_res : search_res) {
+        auto i = find_element_by_name(search, s_res, "CA");
+
         if (i == search.size()) {
             continue;
         }
-        for (const auto& nres : native_res) {
-            if (*sres.id() == *nres.id() &&
-                sres.get("chainname") == nres.get("chainname")) {
-                auto j = find_element_by_name(native, nres, "CA");
-                if (j == native.size()) {
-                    continue;
-                }
 
-                a_search.emplace_back(i);
-                a_native.emplace_back(j);
-                n_ali++;
-                break;
+        auto s_chain_name = s_res.get("chainname")->as_string();
```

Comment:
Since you are relying on the presence of the "chainname" property, I am a bit wary of the wording in the paper that any format supported by chemfiles can be used. Most of them don't set the chainname since the information do not exist.

Reply:
I will change this. I have trouble remembering the `get<Property::STRING>` syntax sometimes.

To clarify, in the paper we do not (or should not) claim that we can read all formats (only write). I will address this separately.

Reply:
> I have trouble remembering the get<Property::STRING> syntax sometimes.

Yeah, this is something I wanted to improve, but never got around